src/spec/sync/product-sync.spec.coffee

```diff
@@ -226,6 +226,120 @@ describe 'ProductSync', ->
+              {id: 'p-8', value: {currencyCode: 'EUR', centAmount: 2100}, country: 'FR'},
+            ]
+          }
+          {
+            id: 3
+            prices: [
+              {id: 'p-10', value: {currencyCode: 'EUR', centAmount: 5889}, country: 'DE'},
+            ]
+          }
+        ]
+
+      newProduct =
+        id: '123'
+        masterVariant:
+          sku: 'v1'
+          prices: [
+            {id: 'p-1', value: {currencyCode: 'GBP', centAmount: 555}},
+            {id: 'p-2', value: {currencyCode: 'GBP', centAmount: 245}},
+          ]
+        variants: [
+          {
+            id: 2
+            prices: [
+              {id: 'p-7', value: {currencyCode: 'USD', centAmount: 4444}},
+              {id: 'p-8', value: {currencyCode: 'USD', centAmount: 5555}, country: 'US'},
+            ]
+          }
+          {
+            id: 3
+            prices: [
+              {id: 'p-10', value: {currencyCode: 'USD', centAmount: 1257}, country: 'US'},
+            ]
+          }
+        ]
+      update = @sync.buildActions(newProduct, oldProduct).getUpdatePayload()
+
+      expected_update =
+        actions: [
+          { action: 'changePrice', priceId: 'p-1', price: {value: {currencyCode: 'GBP', centAmount: 555} }}
+          { action: 'changePrice', priceId: 'p-2', price: {value: {currencyCode: 'GBP', centAmount: 245} }}
+          { action: 'changePrice', priceId: 'p-7', price: {value: {currencyCode: 'USD', centAmount: 4444} }}
+          { action: 'changePrice', priceId: 'p-8', price: {value: {currencyCode: 'USD', centAmount: 5555}, country: 'US' }}
+          { action: 'changePrice', priceId: 'p-10', price: {value: {currencyCode: 'USD', centAmount: 1257}, country: 'US' }}
+        ]
+        version: oldProduct.version
+      expect(update).toEqual expected_update
+
+    it 'should create `changePrice` action if price selection is unique', ->
+      oldProduct =
+        id: '123'
+        version: 1
+        masterVariant:
+          id: 1
+          sku: 'v1'
+          prices: [
+            {id: 'p-1', value: {currencyCode: 'EUR', centAmount: 100}, validUntil: '2019-10-16'},
+            {id: 'p-2', value: {currencyCode: 'EUR', centAmount: 1000}, country: 'DE'},
+            {id: 'p-3', value: {currencyCode: 'GBP', centAmount: 1000}},
+          ]
+        variants: [
+          {
+            id: 2
+            prices: [
+              {id: 'p-8', value: {currencyCode: 'USD', centAmount: 2100}, country: 'US', customerGroup: {id: 'special-price-id', typeId: 'customer-group'}},
+            ]
+          }
+        ]
+
+      newProduct =
+        id: '123'
+        masterVariant:
+          sku: 'v1'
+          prices: [
+            {value: {currencyCode: 'EUR', centAmount: 555}, validUntil: '2020-12-14'},
+            {value: {currencyCode: 'EUR', centAmount: 245}, country: 'DE'},
+            {value: {currencyCode: 'GBP', centAmount: 2300}}
```

Comment:
Thanks for adding a test. Could you please also mix the order of price objects within prices array of `newProduct.masterVariant` as also change order of properties of price object properties itself? Maybe i am paranoid but i have doubts regarding how diffpatch lib is matching/comparing prices as order is not guaranteed from feed and API response and matching should always work.

Reply:
Mixing the order will not trigger a `changePrice` action, but instead a `removePrice` and `addPrice`. This was supposed to be a quick fix and going ahead to check against all prices in the array regardless of their position will introduce too much logic to the code. That and the fact that this is soon-to-be deprecated makes me think it won't be worth the effort

Reply:
Does new nodejs library support matching prices by scope already? If not maybe we should at least create an issue there?

Reply:
The JS SDK doesn't support this. I will add a ticket for that